core/modes/language_modes.py

```diff
@@ -83,6 +83,8 @@
     for ext in extensions.split()
 }
 
+language_ids = set(language_extensions.keys())
```

Comment:
If some other external file needs the list of all language identifiers it's preferable to have that as a separate object instead of having to understand the internal language id to file ending map